server/src/response.rs

```diff
@@ -486,17 +489,83 @@ fn make_stop_point_pt_object(
     let mut proto = navitia_proto::PtObject {
         name: model.stop_point_name(stop_point_idx).to_string(),
         uri: model.stop_point_uri(stop_point_idx),
-        stop_point: Some(make_stop_point(stop_point_idx, model)),
+        stop_point: Some(make_stop_point(stop_point_idx, model, false)),
         ..Default::default()
     };
     proto.set_embedded_type(navitia_proto::NavitiaType::StopPoint);
 
     Ok(proto)
 }
 
+fn make_access_point(
+    model: &ModelRefs,
+    pathway: &Pathway,
+    access_point_idx: StopLocationIdx,
+) -> Option<navitia_proto::AccessPoint> {
+    let is_entrance = pathway.is_bidirectional || pathway.from_stop_type == StopType::StopEntrance;
+    let is_exit = pathway.is_bidirectional || pathway.to_stop_type == StopType::StopEntrance;
+
+    let pathway_stop_point_idx = if pathway.from_stop_type == StopType::Point {
+        model.base.stop_point_idx(&pathway.from_stop_id)?
+    } else if pathway.to_stop_type == StopType::Point {
+        model.base.stop_point_idx(&pathway.to_stop_id)?
+    } else {
+        warn!("Something weired occurred when filling pathway: cannot find stop point");
+        return None;
+    };
+    let coord = model.base.stop_location_coord(access_point_idx);
+
+    Some(navitia_proto::AccessPoint {
+        name: Some(model.base.stop_location_name(access_point_idx).to_string()),
+        uri: Some(format!(
+            "access_point:{}",
+            model.base.stop_location_id(access_point_idx)
+        )),
+        coord: Some(navitia_proto::GeographicalCoord {
+            lat: coord.lat,
+            lon: coord.lon,
+        }),
+        is_entrance: Some(is_entrance),
+        is_exit: Some(is_exit),
+        pathway_mode: Some(convert_pathwaymode(&pathway.pathway_mode)),
+        length: pathway.length.map(|x| {
+            x.floor()
+                .to_i32()
+                .or_else(|| {
+                    warn!("cannot convert pathway length to i32");
+                    Some(0)
+                })
+                .unwrap()
+        }),
+        traversal_time: pathway.traversal_time.map(|x| {
+            i32::try_from(x).expect("traversal_time shouldn't be greater than u32::MAX/2")
+        }),
+        stair_count: pathway.stair_count.map(|x| x as i32),
+        max_slope: pathway.max_slope.map(|x| x as i32),
+        min_width: pathway.min_width.map(|x| x as i32),
```

Comment:
```suggestion
        stair_count: pathway.stair_count.map(|x| i32::from(x)),
        max_slope: pathway.max_slope.map(|x| x.floor().to_i32()).flatten(),
        min_width: pathway.min_width.map(|x| x.floor().to_i32()).flatten(),
```
There is a lot of "hidden" things that may happen behind a "as" cast https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast
So I would rather use more explicit conversion methods